src/prefect/contrib/tasks/databricks/databricks.py

```diff
@@ -0,0 +1,260 @@
+        start_time = time()
+        while time() - start_time < timeout:
+            response = requests.get(
+                url=query,
+                timeout=self._timeout_seconds,
+                headers={
+                    "Authorization": f"Bearer {self._databricks_token}",
+                    "User-Agent": USER_AGENT,
+                },
+            )
+
+            response.raise_for_status()
+            data = response.json()
+            run_page, run_name, lifecycle_state = (
+                data["run_page_url"],
+                data["run_name"],
+                data["state"]["life_cycle_state"],
+            )
+
+            if "result_state" in data["state"]:
+                status = data["state"]["result_state"]
+                self.logger.info(
+                    f"Databricks {run_name} is {status}... For more info please visit {run_page}"
+                )
+                return data
+
+            self.logger.debug(
+                f"Databricks {run_name} is {lifecycle_state}... For more info please visit {run_page}"
+            )
+            sleep(polling_period)
+
+        raise Exception(
+            f"DatabricksRunSubmit timeout ! Unable to fetch {query} in the time imparted"
+        )
+
+    def run(self) -> None:
+        """
+        Task run method
+
+        """
+        attempt_num = 1
+        while attempt_num < self._databricks_retry_limit:
+            self.logger.info(f"{self.name}: try # {attempt_num} ")
+            try:
+                response = requests.post(
+                    url=f"{self._end_point}/jobs/runs/submit",
+                    json=self._json,
+                    headers={
+                        "Authorization": f"Bearer {self._databricks_token}",
+                        "User-Agent": USER_AGENT,
+                    },
+                    timeout=self._timeout_seconds,
+                )
+
+                response.raise_for_status()
+                json_data = response.json()
+                if "run_id" not in json_data:
+                    raise requests.exceptions.RequestException
+
+                result_data = self._wait_for_run_to_complete(
+                    run_id=json_data["run_id"],
+                    polling_period=self._polling_period,
+                    timeout=self._polling_timeout,
+                )
+
+                result_status, run_name, run_page = (
+                    result_data["state"]["result_state"],
+                    result_data["run_name"],
+                    result_data["run_page_url"],
+                )
+
+                if result_status != "SUCCESS":
+                    raise Exception(
+                        f"Task {run_name} failed with status {result_status}... "
+                        f"For more info please visit {run_page}"
+                    )
+                return
+
+            except requests.exceptions.RequestException as error:
+                self.logger.error(error)
```

Comment:
we should log how many attempts we have done here and whether or not we are going to  retry.